embedded-graphics/src/primitives/ellipse_quadrant.rs

```diff
@@ -0,0 +1,434 @@
+use crate::draw_target::DrawTarget;
+use crate::drawable::Drawable;
+use crate::drawable::Pixel;
+use crate::geometry::Dimensions;
+use crate::geometry::{Point, Size};
+use crate::pixelcolor::PixelColor;
+use crate::primitives::ellipse::Ellipse;
+use crate::primitives::ellipse::StyledEllipseIterator;
+use crate::primitives::rectangle::{self, Rectangle};
+use crate::primitives::ContainsPoint;
+use crate::primitives::Primitive;
+use crate::style::PrimitiveStyle;
+use crate::style::Styled;
+
+/// A quadrant around an origin
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum Quadrant {
+    TopLeft = 0,
+    TopRight = 1,
+    BottomRight = 2,
+    BottomLeft = 3,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub(crate) struct EllipseQuadrant {
+    pub radius: Size,
+    pub quadrant: Quadrant,
+    pub bounding_box: Rectangle,
+    pub ellipse: Ellipse,
+}
+
+impl EllipseQuadrant {
+    pub fn new(top_left: Point, radius: Size, quadrant: Quadrant) -> Self {
+        let ellipse_top_left = match quadrant {
+            Quadrant::TopLeft => top_left,
+            Quadrant::TopRight => top_left - radius.x_axis(),
+            Quadrant::BottomRight => top_left - radius,
+            Quadrant::BottomLeft => top_left - radius.y_axis(),
+        };
+
+        Self {
+            radius,
+            ellipse: Ellipse::new(ellipse_top_left, radius * 2),
+            quadrant,
+            bounding_box: Rectangle::new(top_left, radius),
+        }
+    }
+
+    /// Expand the ellipse quadrant by a given offset
+    ///
+    /// This method will expand the ellipse quadrant in the direction of its outer edge. The center
+    /// point will not be moved.
+    ///
+    /// If the radius of the ellipse quadrant is zero, no expand will be performed.
```

Comment:
Lots of caveats here, but this is the exact behaviour I need for `RoundedRectangle`. Should it be made more general with a `RoundedRectangle`-specific method somewhere else?

Reply:
If this primitive is only used internally we don't need a generic `expand` impl. But I would change the name to make it clear that this function is different than the other `expand` methods.